consensus/src/experimental/buffer_manager.rs

```diff
@@ -164,6 +167,60 @@ impl StateManager {
         }
     }
 
+    async fn process_ordered_blocks(
+        &mut self,
+        ordered_blocks: OrderedBlocks,
+    ) -> anyhow::Result<()> {
+        let OrderedBlocks {
+            blocks,
+            finality_proof,
+            callback,
+        } = ordered_blocks;
+
+        // push blocks to buffer
+        for eb in blocks.iter() {
```

Comment:
do we need to split it into single block?

Reply:
I thought it might be good to track the fine-grained execution progress to avoid unnecessary retries when there is a failure.